hydromt/data_catalog.py

```diff
@@ -1472,7 +1477,6 @@ def get_geodataframe(
                 bbox,
                 buffer,
                 predicate,
-                handle_nodata,
```

Comment:
Because handle_nodata is not done here, function should return gdf rather than already return out of the function?
ie gdf = GeoDataFrameAdapter._slice_data? Or at least the exex_nodata_strategy before return?
I'm guessing the following get_data call could go in the if isinstance(data_like, (str. Path)) above for more clarity?

Reply:
I think this comment is the same for a lot of get_* methods (at least not true for get_dataframe but true for get_rasterdataset). That in case a python object is passed, the NoDataStrategy is not applied.

Reply:
I don't think I understand this comment. I thought that was what we described before, that if the user passes in a data container themselves then we don't have to do the no_data handeling? so that we only have to handle the no data in the get data. Could you elaborate on what you mean here? (we can also quickly call about it if you think that's easier)

Reply:
So if the user passes a geodataframe, it could still be that after slicing an empty gdf is returned. So after slicing, you may still want to do the if has_nodata check and then return None or error in case the data is empty.